typescript/appsync-graphql-dynamodb/index.ts

```diff
@@ -44,28 +44,38 @@ export class AppSyncCdkStack extends cdk.Stack {
       }`
     });
 
-    const itemsTable = new Table(this, 'ItemsTable', {
+    const itemsTable = new CfnTable(this, 'ItemsTable', {
       tableName: tableName,
-      partitionKey: {
-        name: `${tableName}Id`,
-        type: AttributeType.STRING
-      },
+      keySchema: [
+          {
+            attributeName: `${tableName}Id`,
+            keyType: "HASH",
+          }
+      ],
+      attributeDefinitions: [
+          {
+            attributeName: `${tableName}Id`,
+            attributeType: AttributeType.STRING,
+          }
+      ],
       billingMode: BillingMode.PAY_PER_REQUEST,
-      stream: StreamViewType.NEW_IMAGE
+      // streamSpecification: StreamViewType.NEW_IMAGE,
     });
+    itemsTable.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.DELETE 
```

Comment:
not in the [aws-cdk docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-dynamodb.CfnTable.html), the default for `CfnTable` is `cdk.CfnDeletionPolicy.RETAIN`, set to DELETE for testing

Reply:
instead of using `CfnTable` construct, you could use `Table` itself. Then to apply `deletionPolicy`, use TypeScript magic :)
```
(<CfnTable>(<any> itemsTable).table).cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.DELETE;
```

Reply:
@cvrajeesh that's a good suggestion, the reason I'm using `CfnTable` here is for adding a `dependsOn` for `apiSchema` here: https://github.com/aws-samples/aws-cdk-examples/blob/dfec54b7ac9d2932301c86db9386a649b73868fb/typescript/appsync-graphql-dynamodb/index.ts#L71

Reply:
Forgot to also mention the reason for the `.addDependsOn` is that so the stack didn't fail to stand up - since `ApiSchema` references the `tableName` (through locals on the `tableName` var) it still fails to stand up without the dependsOn here.

Reply:
> instead of using `CfnTable` construct, you could use `Table` itself. Then to apply `deletionPolicy`, use TypeScript magic :)
> 
> ```
> (<CfnTable>(<any>itemsTable).table).cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.DELETE;
> ```

Using the CFN escape-hatch should be a last resort for CDK examples.

Reply:
makes sense.